### PR TITLE
macOS 10.10 and SDK 10.11 minimum

### PR DESCRIPTION
**Pull request by csomor**
All changes related to raising the minimum requirements

### CODE REVIEW

**Comment by csomor** [comment]
@vadz if you want you could also push your changes regarding 10.11 SDK into this PR

**Comment by MaartenBent** [comment]
Hi, there is also some code in CMake that can be updated
<details>
  <summary>Click for diff</summary>

```diff
 CMakeLists.txt              | 2 +-
 build/cmake/functions.cmake | 8 --------
 build/cmake/init.cmake      | 5 -----
 build/cmake/setup.cmake     | 6 +-----
 4 files changed, 2 insertions(+), 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c671fb1c0b..ac688cbfb8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,7 @@ endif()
 if(APPLE AND NOT CMAKE_OSX_DEPLOYMENT_TARGET)
     # If no deployment target has been set default to the minimum supported
     # OS X version (this has to be set before the first project() call)
-    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.9 CACHE STRING "OS X Deployment Target")
+    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.10 CACHE STRING "OS X Deployment Target")
 endif()
 
 include(build/cmake/policies.cmake NO_POLICY_SCOPE)
diff --git a/build/cmake/functions.cmake b/build/cmake/functions.cmake
index 741358f633..c33dd3946c 100644
--- a/build/cmake/functions.cmake
+++ b/build/cmake/functions.cmake
@@ -77,14 +77,6 @@ endmacro()
 function(wx_set_common_target_properties target_name)
     cmake_parse_arguments(wxCOMMON_TARGET_PROPS "DEFAULT_WARNINGS" "" "" ${ARGN})
 
-    if(APPLE AND CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS 10.9 AND wxHAS_CXX11)
-        if(CMAKE_GENERATOR STREQUAL "Xcode")
-            set_target_properties(${target_name} PROPERTIES XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY libc++)
-        else()
-            target_compile_options(${target_name} PUBLIC "-stdlib=libc++")
-            target_link_libraries(${target_name} PRIVATE "-stdlib=libc++")
-        endif()
-    endif()
     set_target_properties(${target_name} PROPERTIES
         LIBRARY_OUTPUT_DIRECTORY "${wxOUTPUT_DIR}${wxPLATFORM_LIB_DIR}"
         ARCHIVE_OUTPUT_DIRECTORY "${wxOUTPUT_DIR}${wxPLATFORM_LIB_DIR}"
diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
index 4524bc70e6..d013e92a5a 100644
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -11,11 +11,6 @@
 if(DEFINED wxBUILD_CXX_STANDARD AND NOT wxBUILD_CXX_STANDARD STREQUAL COMPILER_DEFAULT)
     set(CMAKE_CXX_STANDARD ${wxBUILD_CXX_STANDARD})
 endif()
-if(NOT CMAKE_CXX_STANDARD EQUAL 98)
-    set(wxHAS_CXX11 TRUE)
-else()
-    set(wxHAS_CXX11 FALSE)
-endif()
 
 if(MSVC)
     # Determine MSVC runtime library flag
diff --git a/build/cmake/setup.cmake b/build/cmake/setup.cmake
index e26af41982..1272e3de49 100644
--- a/build/cmake/setup.cmake
+++ b/build/cmake/setup.cmake
@@ -23,10 +23,6 @@ include(CheckTypeSize)
 include(CMakePushCheckState)
 include(TestBigEndian)
 
-if(APPLE AND CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS 10.9 AND wxHAS_CXX11)
-    set(CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} "-stdlib=libc++")
-endif()
-
 # Add a definition to setup.h and append it to a list of defines for
 # for compile checks
 macro(wx_setup_definition def)
@@ -143,7 +139,7 @@ wx_check_c_source_compiles(
     stdio.h
     )
 #TODO: wxNO_VARIADIC_MACROS
-if(wxUSE_STL AND NOT wxHAS_CXX11)
+if(wxUSE_STL AND CMAKE_CXX_STANDARD EQUAL 98)
     wx_check_cxx_source_compiles("
         std::vector<int> moo;
         std::list<int> foo;

```

</details>

**Comment by csomor** [comment]
> Hi, there is also some code in CMake that can be updated

thanks a lot @MaartenBent

File: docs/changes.txt
Comment (vadz) [inline docs/changes.txt:156]: Just a nitpick, but this wording could IMHO be improved:
```suggestion
  higher is supported, you must use at least Xcode 7.2.1.
```

File: .travis.yml
Comment (vadz) [inline .travis.yml:37]: Not directly related to your changes, I wonder if we should remove C++11 options from this one, just to keep testing C++98 as long as we claim to continue supporting it (all the other Travis CI builds use C++11 or even C++17). Or maybe Xcode 7.3 compiler already uses C++11 by default anyhow? I don't know where can I find this information about Apple clang...

**Review by vadz — APPROVED** [review]
Thanks!

I'll make configure changes and merge this one once the CI builds pass if you don't mind.

File: .travis.yml
Comment (csomor) [reply .travis.yml:37]: I think C++98 is default behavior, I have found two tables, so to me it looks like all C++11 and C++14 features should be working

https://trac.macports.org/wiki/XcodeVersionInfo
https://clang.llvm.org/cxx_status.html

so if we have this Xcode version as a minimum, why would we have to support C++98 on macOS ?

Comment (vadz) [reply .travis.yml:37]: Actually I've just checked and the default is still C++98 even in Xcode 10.2 (which I currently have), strangely enough.

As for supporting C++98, I'm fine with not supporting it any longer under macOS, but then we need to enable C++11 in configure automatically and document that C++98 is not supported any more. It's also somewhat inconsistent to not support it under macOS while we still support it under Linux/other Unices, I'd rather drop it everywhere at once.

Comment (csomor) [reply .travis.yml:37]: I can see the point there, but I don't want to wait too long;-)
Is it on other platforms the same situation that all the compilers we support, also support c++11 ? so that we don't loose a platform if we decide this ?

**Comment by csomor** [comment]
> I'll make configure changes and merge this one once the CI builds pass if you don't mind.

that's fine, thanks

File: .travis.yml
Comment (vadz) [reply .travis.yml:37]: No, unfortunately it's still not the case. There are CentOS/RHEL 6 systems (which I have to target...) that don't have C++11 compiler out of the box and only 4.8 (which doesn't have great C++11 support) can be easily installed on them. Also some non-desktop platforms, but wx is probably not too relevant there (although sometimes it's still used). And, of course, there are still all the MSVS versions earlier than 2013, which was the first one to have support for a decent subset of C++11, that we still support. I'm afraid C++98 will still be supported in 3.2.0.

BTW, do we still need `-stdlib=libc++` with the (now) supported Xcode versions or can I remove the part adding it from configure?

Comment (csomor) [reply .travis.yml:37]: for Xcode itself we don't need it, once you are above or at macos-min 10.9, I'd assume command line builds do likewise, I'll check

Comment (csomor) [reply .travis.yml:37]: > I'm afraid C++98 will still be supported in 3.2.0.

So then I'd really like to have at least in the macOS code, as we don't loose any systems.

**Comment by vadz** [comment]
I think that's all that needs to be changed at configure level, but please let me know if there is anything else (and also if you see any improvements to the error messages that would make them more clear for people speaking Macish). TIA!

**Comment by csomor** [comment]
> I think that's all that needs to be changed at configure level, but please let me know if there is anything else (and also if you see any improvements to the error messages that would make them more clear for people speaking Macish). TIA!

Thanks a lot, the only thing that comes to my mind is when telling them to use a different min os or sdk, to also give them our precise option name for the configure line ?
and a small nitpick regarding the command line tools: the minimum Xcode is 7.2.1 only that one runs on 10.10.5, only on travis we must use 7.3 (which only runs on 10.11+) because 7.2.1 is not available on travis

**Comment by vadz** [comment]
> Thanks a lot, the only thing that comes to my mind is when telling them to use a different min os or sdk, to also give them our precise option name for the configure line ?

I thought about this but decided that if they used the option in the first place (and they have to, in order for this mismatch to happen), they must presumably know what it is called. And the error message is already relatively long... But now that I think about it, perhaps I could split it over multiple lines. Let me experiment with this...

> and a small nitpick regarding the command line tools: the minimum Xcode is 7.2.1 only that one runs on 10.10.5, only on travis we must use 7.3 (which only runs on 10.11+) because 7.2.1 is not available on travis

Corrected, thanks!

I'll postpone this commit until the CI builds pass, and will merge them directly without another round of rebuilding.

**Comment by csomor** [comment]
one thing, which I'm not sure of, are we somewhere also testing against the minimum deployment ? wxUSE_MACOSX_VERSION_MIN shouldn' t be less than 10.10, if this is not easy, we could also do it along the same idea as the SDK check, but this time using `__MAC_OS_X_VERSION_MIN_REQUIRED` which should be `>= 101000`

**Comment by vadz** [comment]
Thanks, I've added min version check and also removed `-stdlib=libc++` option, so finally we do need to retest the build and so I've pushed my changes.

File: CMakeLists.txt
Comment (MaartenBent) [inline CMakeLists.txt:32]: This one also has `OS X` instead of `macOS`.
```suggestion
    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.10 CACHE STRING "macOS Deployment Target")
```

Comment (csomor) [reply CMakeLists.txt]: committed, thanks

**Comment by vadz** [comment]
I've finally merged this without waiting for the CI builds as the branch couldn't be merged automatically anyhow any longer, due to the conflict in `.travis.yml`, hopefully my last change repaired wxiOS build breakage -- if not, I'll fix it a.s.a.p. in master.

**Pull request merged**
